### PR TITLE
fix(BaseInteraction): add missing props

### LINKED ISSUE #10516: AutocompleteInteraction does not define context or authorizingIntegrationOwners
### Which package is this bug report for?

discord.js

### Issue description

According to [Discord documentation](https://discord.com/developers/docs/interactions/receiving-and-responding), `authorizing_integration_owners` is available for all interactions like `id` or `type`. `context` is nullable but nothing suggests that it is not present for autocomplete interactions as a rule. In Discord.js, these fields are only available for [CommandInteraction](https://discord.js.org/docs/packages/discord.js/stable#/docs/discord.js/main/class/CommandInteraction?scrollTo=authorizingIntegrationOwners) and not [AutocompleteInteraction](https://discord.js.org/docs/packages/discord.js/stable#/docs/discord.js/main/class/AutocompleteInteraction).

### Code sample

```typescript
function onCommand(interaction: ChatInputCommandInteraction) {
	interaction.authorizingIntegrationOwners;
	interaction.context;
}

function onAutocomplete(interaction: AutocompleteInteraction) {
	interaction.authorizingIntegrationOwners; // Property 'authorizingIntegrationOwners' does not exist on type 'AutocompleteInteraction<CacheType>'.ts(2339)
	interaction.context; // Property 'context' does not exist on type 'AutocompleteInteraction<CacheType>'.ts(2339)
}
```

### Versions

discord.js 14.16.2

### Issue priority

Low (slightly annoying)

### Which partials do you have configured?

Not applicable

### Which gateway intents are you subscribing to?

Not applicable

### I have tested this issue on a development release

_No response_

### PR DESCRIPTION
**Pull request by monbrey**
Fixes #10516 

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Comment by kevinlul** [comment]
I know I reported for specifically AutocompleteInteraction because that's the type affecting our bot, but this implementation may need to be moved up to BaseInteraction if all interactions are supposed to have these fields.

**Review by Jiralite — APPROVED** [review]

**Comment by monbrey** [comment]
> I know I reported for specifically AutocompleteInteraction because that's the type affecting our bot, but this implementation may need to be moved up to BaseInteraction if all interactions are supposed to have these fields.

Good point - I had thought it would only apply to "installed" commands, but it does seem to be on all of them. I'll move it.

File: packages/discord.js/src/structures/BaseInteraction.js [hunk 1]
```diff
@@ -107,6 +107,21 @@ class BaseInteraction extends Base {
...
+     * @see {@link https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-authorizing-integration-owners-object}
+     */
+    this.authorizingIntegrationOwners = data.authorizing_integration_owners;
+    /* eslint-enable max-len */
+
+    /**
+     * Context where the interaction was triggered from
+     * @type {?InteractionContextType}
+     */
+    this.context = data.context ?? null;
```
Comment (TAEMBO) [inline packages/discord.js/src/structures/BaseInteraction.js:124]: The initialization of these properties should be removed from the CommandInteraction constructor if BaseInteraction is now handling the initialization of them.

File: packages/discord.js/typings/index.d.ts [hunk 2]
```diff
@@ -1203,6 +1201,7 @@ export class AutocompleteInteraction<Cached extends CacheType = CacheType> exten
   public commandName: string;
   public commandType: ApplicationCommandType.ChatInput;
   public commandGuildId: Snowflake | null;
+  public context: InteractionContextType | null;
```
Comment (sdanialraza) [inline packages/discord.js/typings/index.d.ts:1204]:
```suggestion
```

**Review by vladfrangu — APPROVED** [review]

**Review by didinele — APPROVED** [review]

**Pull request merged**